docs/_source/docs/template_handlers.rst

```diff
@@ -0,0 +1,185 @@
+Template Handlers
+=================
+
+Template handlers can be used to resolve a ``template`` config block to a CloudFormation template. This can be used to
+fetch templates from S3, for example. Users can create their own template handlers to easily add support for other
+template loading mechanisms. See `Custom Template Handlers`_ for more information.
+
+When a ``template_path`` property is specified in the Stack config, it is wired into a ``file`` template handler by
+default. This saves you from having to specify a full ``template`` block if you just want to load a file from disk.
+Sceptre implements resolvers, which can be used to resolve a value of a
+CloudFormation ``parameter`` or ``sceptre_user_data`` value at runtime. This is
+most commonly used to chain the outputs of one Stack to the inputs of another.
+
+Syntax:
+
+.. code-block:: yaml
+
+   template:
+     type: s3
+     path: <bucket>/<key>
+
+Available Template Handlers
+---------------------------
+
+file
+~~~~~~~~~~~~~~~~~~~~
+
+Loads a template from disk. Supports JSON, YAML, Jinja2 and Python files. Will be used if the ``template_path`` Stack
+config property is set, for backwards compatibility reasons.
+
+Syntax:
+
+.. code-block:: yaml
+
+   template:
+     type: file
+     path: <path>
+
+Example:
+
+.. code-block:: yaml
+
+   template:
+     type: file
+     path: storage/bucket.yaml
+
+s3
+~~~~~~~~~~~~~
+
+Downloads a template from an S3 bucket. The template will be kept in memory and not be persisted to disk. Currently
+only supports YAML / JSON CloudFormation templates,
+
+Syntax:
+
+.. code-block:: yaml
+
+   template:
+     type: s3
+     path: <bucket>/<key>
+
+Example:
+
+.. code-block:: yaml
+
+   template:
+     type: s3
+     path: infra-templates/s3/v1/bucket.yaml
+
+Custom Template Handlers
+------------------------
+
+If you need to load templates from a different source, you can write your own template handler.
+
+A template handler is a Python class which inherits from abstract base class ``TemplateHandler`` found in the
+``sceptre.template_handlers`` module.
+
+To have Sceptre validate that the ``template`` block specified in the Stack config is correct, template handlers
+should provide a JSON schema with the required and optional properties. The ``schema()`` method should be
+implemented and return a Python dictionary with the schema. For examples of JSON schemas in Python, please see
+the documentation of the `jsonschema library`_.
+
+Template handlers get access to the ``template`` block parameters, ``sceptre_user_data`` and ``connection_manager``.
+These properties are available on ``self``. Using ``connection_manager``, template handlers can call AWS SDKs
```

Comment:
should `AWS SDKs` be `AWS python SDKs` or `AWS boto3 APIs` and provide a link to the boto3 libs?

Reply:
I updated the wording to say `AWS endpoints` and added a hint that they correspond to the Boto3 services / commands.